### ISSUE TITLE
[BUG][mthreads]  error: ran out of registers during register allocation:    TEMP_4096_with_sub0...  _n_sub_lo16_in_RC_INT_SRC2_H

### ISSUE THREAD

**Issue by O1iveira7** [FlagTree/flagtree#1008]
## Environment
  - Triton 3.6.0 (FlagTree mthreads3.6), MUSA backend, MTT S5000
  - torch_musa 2.7.1, MUSA 4.3.5, Python 3.10
## Symptom
 Compiling a valid Triton kernel (2D max-pooling, 42_2d_max_pooling) fails deterministically in the MTGPU backend's llc with:
   ```
     error: ran out of registers during register allocation:
   TEMP_4096_with_sub0_sub1_..._in_RC_DATA_QQQ_with_sub32..sub63_in_RC_DATA_DQQ_.
   ..
   ```
The kernel is semantically equivalent to the original (which compiles and runs fine under the exact same launch parameters): the only difference is one semantics-irrelevant scalar store (tl.store(output + 26, 1.0)) inserted mid-kernel. The failure is a compile-time robustness issue of the MTGPU llc register allocator, not invalid Triton code.

**FlagTree mthreads3.6 output:**
> COMPILE FAILED: `llc` failed with error code 1
Expected: compile succeeds; the inserted store is a scalar store to an unused  location and must not change codegen validity.
**Triton3.7 RTX3070 output:**
> COMPILE+LAUNCH OK
> EXIT=0

##  Details / root-cause analysis

   • The original kernel (without the injected store) compiles and runs fine with the same parameters
     (BLOCK_SIZE_OUTPUT=1024, num_warps=4).
   • The failure is a statement-position / codegen-shape effect, not a semantic one:
       • inserting the scalar store between input_mask_h and input_mask_w triggers it (3/3);
       • merely swapping two unrelated statements (no store at all) does not trigger;
       • the store's target pointer, value and mask do not matter (probe experiments).
   • LLVM O3 (inside make_llir) keeps the integer division sdiv → mulhi lowering vectorized (<64 x i64>,
     zext+mul+lshr+trunc, 4096-bit SSA values) when the store is present; without it the chain is scalarized. The wide
     values are then assigned to the RC_DATA_QQQ / B4096 register class by MTGPU PreRA, which cannot fit a 4096-bit value
     into the physical register file → deterministic error.
   • Sensitivity: triggers only at BLOCK_SIZE_OUTPUT=1024 with num_warps=4 (256 elements/warp). num_warps=2/8 or
     BLOCK_SIZE_OUTPUT=512/2048 compile fine. There is no monotonic resource-pressure dependence.

  ## Minimal reproducer
```
    import torch
     import triton
     import triton.language as tl


     @triton.jit
     def max_pooling_kernel(
         input: tl.tensor, output: tl.tensor,
         input_size_h: int, input_size_w: int,
         output_size_h: int, output_size_w: int,
         kernel_size: int, stride: int, padding: int,
         BLOCK_SIZE_KERNEL: tl.constexpr, BLOCK_SIZE_OUTPUT: tl.constexpr,
     ):
         pid = tl.program_id(0)
         slice = tl.program_id(1)
         output_offset = pid * BLOCK_SIZE_OUTPUT + tl.arange(0,
   BLOCK_SIZE_OUTPUT)
         output_idx_h = output_offset // output_size_w
         output_idx_w = output_offset % output_size_w
         input_idx_h = output_idx_h[:, None] * stride + (tl.arange(0,
   BLOCK_SIZE_KERNEL) - padding)[None, :]
         input_idx_w = output_idx_w[:, None] * stride + (tl.arange(0,
   BLOCK_SIZE_KERNEL) - padding)[None, :]
         input_ptrs = input_idx_h[:, :, None] * input_size_w + input_idx_w[:,
   None, :]
         input_mask_h = (input_idx_h >= 0) & (input_idx_h < input_size_h)
         tl.store(output + 26, 1.0)  # <<< the only change vs the original kernel
         input_mask_w = (input_idx_w >= 0) & (input_idx_w < input_size_w)
         input_mask = input_mask_h[:, :, None] & input_mask_w[:, None, :]
         kernel_mask = tl.arange(0, BLOCK_SIZE_KERNEL) < kernel_size
         kernel_mask = kernel_mask[:, None] & kernel_mask[None, :]
         input_ptrs = input_ptrs + slice * input_size_h * input_size_w
         input_data = tl.load(input + input_ptrs, mask=input_mask & kernel_mask,
   other=-float("inf"))
         output_data = tl.max(
             tl.reshape(input_data, (BLOCK_SIZE_OUTPUT, BLOCK_SIZE_KERNEL *
   BLOCK_SIZE_KERNEL)),
             axis=1, keep_dims=False,
         )
         output_mask = output_offset < output_size_h * output_size_w
         output_ptr = output_offset + slice * output_size_h * output_size_w
         tl.store(output + output_ptr,
   output_data.to(dtype=output.dtype.element_ty), mask=output_mask)


     input_t = torch.randn(50, device="musa")          # N=1 C=2 H=5 W=5
     output = torch.empty(50, device="musa")
     try:
         max_pooling_kernel[(1, 2)](                  # grid=(cdiv(25,1024),
   N*C)=(1,2)
             input_t, output, 5, 5, 5, 5, 3, 1, 1,    # H, W, osh, osw, ks,
   stride, padding
             BLOCK_SIZE_KERNEL=4, BLOCK_SIZE_OUTPUT=1024,
         )
         torch.musa.synchronize()
         print("COMPILE+LAUNCH OK")
     except Exception as e:
         print("COMPILE FAILED:", str(e).splitlines()[0])
         print("has signature:", "ran out of registers" in str(e))
 ```

**Comment by zhzhcookie** [comment]
Please try `flagtree===0.6.2a3+mthreads3.6`, or build flagtree from source.
